File: mobile/src/main/java/com/alexstyl/specialdates/upcoming/widget/list/UpcomingEventViewBinder.java
package com.alexstyl.specialdates.upcoming.widget.list;

import android.widget.RemoteViews;

import com.alexstyl.specialdates.upcoming.UpcomingRowViewModel;

public interface UpcomingEventViewBinder<T extends UpcomingRowViewModel> {

    void bind(T viewModel);

    RemoteViews getViews();
}

File: android_mobile/src/main/java/com/alexstyl/specialdates/upcoming/widget/list/UpcomingEventViewBinder.kt
package com.alexstyl.specialdates.upcoming.widget.list

import android.widget.RemoteViews

import com.alexstyl.specialdates.upcoming.UpcomingRowViewModel

interface UpcomingEventViewBinder {

    val views: RemoteViews

    fun bind(viewModel: UpcomingRowViewModel)
}
